mlflow/langchain/__init__.py

```diff
@@ -359,7 +359,7 @@ def load_retriever(persist_directory):
         **model_data_kwargs,
     )
 
-    if Version(langchain.__version__) >= Version("0.0.311"):
+    if Version(langchain.__version__) >= Version("0.0.311") and mlflow_model.resources is None:
```

Comment:
Thoughts on merging the explictly defined resources with the detected resources vs. making the user go one way or the other?

Reply:
Discussed offline with @epec254 that extra_resources is a good idea and can be done later on :)